File: JavaScript/Codewars/Count_the_digit.js









const nbDig = (n, d) => {
  let count = 0
  let squaredNumbers = []
  for (let i = 0; i <= n; i++) {
    squaredNumbers.push(i * i)
  }
  let string = squaredNumbers.join('')
  let array = string.split('')
  array.forEach(digit => {
    if (parseInt(digit) === d) {
      count++
    }
  })
  return count
}

File: Ruby/Codewars/Count_the_digit.rb
def nb_dig(n, d)
	count = 0
  squared_numbers = []
  i = 0
  while i <= n
  squared_numbers << (i ** 2)
  i += 1
  end
  array = squared_numbers.join('').split('')
  array.each do |digit|
    if digit.to_i == d 
      count += 1
    end
  end
  count
end
